docs/DeveloperGuide.md

```diff
@@ -614,6 +653,81 @@ The predicate is then used to filter the list of person.
+* `Money` and its subclass facilitates the payment mechanism. Each `Lesson` contains a `LessonRates` field and a `OutstandingFees` field. Both of which extends `Money`.
+
+When the user launches TAB, in `MainApp#initModelManager()`, the `FeesCalculator` would update all the outstanding lesson fees in the model. Figure I.6.1 is a sequence diagram after user launches TAB.
+
+![UpdateFeesSequenceDiagram](images/UpdateFeesSequenceDiagram.png)
+
+*Figure I.6.1: Sequence diagram of Update Fees.*
+
+In `initModelManager()`, after the `model` was built from storage, `FeesCalculator#UpdateAllLessonOutstanding()` is called to update the lesson fees. For each person in `addressBook`, a new `updatedPerson` would be created.
+When creating the new `updatedPerson`, for each lesson that the person has, `updateLessonOutstandingFeesField()` will update all the outstanding fees fields using `lastUpdatedDate` and the current local date time.  
+
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** 
+FeesCalculator do not refresh the outstanding fees immediately after the lesson ends if TAB is open. Users need to relaunch TAB to see the updated outstanding fees. </div>
+
+
+The following is the logic of `FeesCalculator` to decide whether to update the specific lesson:
+* To construct a `FeesCalculator` object, the `lastUpdatedDate` and `currentDateTime`, which is simply the `LocalDateTime.now()`, are required. 
+* `FeesCalculator` would count the number of lessons that have ended between `lastUpdatedDate` and `currentDateTime` at the time of initialization.
+  * **For Makeup Lessons:** 
+    * Use `Date#getLocalDate()` to get the date of the lesson and `TimeRange#getEnd()` to get the end time of the lesson. Then check for these conditions:
+      * Lesson end date and time is after `lastUpdatedDate`.
+      * Lesson end date and time is before `currentDateTime`.
+      
+    If both conditions are true, update the fees by adding the cost of the lesson to the current outstanding fees.
+  * **For Recurring Lessons:** 
+    * Find the `laterStart` by comparing `startDate` of lesson and `lastUpdatedDate`. This is to handle cases where lesson starts after the last time TAB was launched.
+    * Find the `earlierEnd` by comparing `endDate` of lesson and `currentDateTime`. This is to handle cases where lesson ends before the current launching of TAB.
+    * Calculate the number of lessons that have passed between the `laterStart` and `earlierEnd`.
+    * Check if any of the cancelled dates falls between `laterStart` and `earlierEnd`. If true, deduct the number of lessons accordingly.
+    * Check for the cases in which the day of `lastUpdatedDate` and `currentDateTime` falls on the day of the lesson and deduct accordingly.
+* Multiply the number of lessons calculated with the cost per lesson to get the amount to be added to `OutstandingFees` field.
+
+![UpdateFeesActivityDiagram](images/UpdateFeesActivityDiagram.png)
```

Comment:
Could introduce the diagram first, right now i dont know what i'm looking at, why does it start at "copy..."?